source/frontend/src/features/properties/map/project/add/ProductsArrayForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { LinkButton, StyledRemoveLinkButton } from 'components/common/buttons';
+import GenericModal from 'components/common/GenericModal';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { Section } from 'features/mapSideBar/tabs/Section';
+import { FieldArray, FieldArrayRenderProps, FormikProps, useFormikContext } from 'formik';
+import { useProductProvider } from 'hooks/providers/useProductProvider';
+import React, { useRef, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaTrash } from 'react-icons/fa';
+import styled from 'styled-components';
+
+import { ProductForm, ProjectForm } from './models';
+import ProductSubForm from './ProductSubForm';
+
+export interface IProductsArrayFormProps {
+  field: string;
+  formikProps: FormikProps<ProjectForm>;
+}
+
+export const ProductsArrayForm: React.FunctionComponent<IProductsArrayFormProps> = ({
+  field,
+  formikProps,
+}) => {
+  const arrayHelpersRef = useRef<FieldArrayRenderProps | null>(null);
+  const { values } = useFormikContext<ProjectForm>();
+
+  const [showModal, setShowModal] = useState(false);
+  const [showFileModal, setShowFileModal] = useState(false);
+  const [rowToDelete, setRowToDelete] = useState<number | undefined>(undefined);
+
+  const { retrieveProductFile, retrieveProductFileLoading } = useProductProvider();
+
+  const products = values.products || [];
+
+  const handleRemove = async (index: number) => {
+    const productId = products[index].id;
+    if (productId !== undefined && productId !== 0) {
+      const file = await retrieveProductFile(productId);
+      if (file !== undefined && file !== null) {
+        setShowFileModal(true);
+      } else {
+        showDeleteWarning(index);
+      }
+    } else {
+      showDeleteWarning(index);
+    }
+  };
+
+  const showDeleteWarning = (index: number) => {
+    setShowModal(true);
+    setRowToDelete(index);
+  };
+
+  const confirmDelete = () => {
+    if (rowToDelete !== undefined) {
+      arrayHelpersRef.current?.remove(rowToDelete);
+    }
+    setShowModal(false);
+  };
+
+  return (
+    <>
+      <Section header="Associated products">
+        <LoadingBackdrop show={retrieveProductFileLoading}></LoadingBackdrop>
+        <FieldArray
+          name={field}
+          render={arrayHelpers => {
+            arrayHelpersRef.current = arrayHelpers;
+            return (
+              <>
+                {products.map((product, index, array) => (
+                  <div key={`product-${index}`}>
```

Comment:
can we use something other then the index here? React doesn't recommend using the index unless there is no way to form a stable key:
https://reactjs.org/docs/lists-and-keys.html#keys
I think we could at least use both the id and the index to make keys stable if they do have an id right?

Reply:
No ID at this point :( and the other fields could also be empty. Could generate a GUID or prefix it with -create-